frontend/src/hooks/useStore.ts

```diff
@@ -33,7 +33,7 @@ type State = {
     setHoverCard: (card: CardTypeWithId | null) => void,
     addCardToDeck: (id: string, location: string, cardnumber: string, type: string) => void,
     deleteFromDeck: (id: string) => void,
-    saveDeck: (name: string, navigate: NavigateFunction) => void,
+    saveDeck: (name: string) => void,
```

Comment:
If this is for fetching logic, you could refactor the name to `useFetch`.

In general having multiple state stores might be considered bad practice, since one is for your current GameState it maybe doesn't matter. BUT if you start connecting values between your 2 stores you may run into issues, reevaluate if using 2 stores is the right approach here